test/test_package_identification_ros.py

```diff
@@ -54,8 +54,7 @@ def test_identify():
 
         augmentation_extension.augment_packages([desc])
         assert desc.metadata['version'] == '0.0.0'
-        # TODO: This is a behavior deviation from colcon-core
-        # assert set(desc.dependencies.keys()) == {'build', 'run', 'test'}
+        assert set(desc.dependencies.keys()) == {'build', 'run', 'test'}
```

Comment:
Based on the PR description and the removal of this TODO it seems like the colcon-core behavior here changed. Is it worth cross-referencing the PR where we changed it in this PR description?

Reply:
This change aligns the behavior with how it has always been in `colcon-core`. Until this change, the behavior has always been different in `colcon-ros`, which I view as a bug. I left the TODO in there when I ported the tests from `colcon-core` to `colcon-ros`.

It's pretty inconsequential though - all we're doing is ensuring that empty lists get added to the dependency dict for all of the supported categories regardless of what dependencies are actually defined by the package. In theory, we could use this information elsewhere to get an idea of what dependency categories we looked for when the package was inspected, as opposed to which ones we actually found.